scripts/appsubmitter.py

```diff
@@ -0,0 +1,176 @@
+        name (str): The name of the training materials.
+        description (str): The description of the training materials.
+        tags (list): The tags associated with the training materials.
+        type_ (list): The type of the training materials.
+        url (str): The URL of the training materials.
+
+    Returns:
+        pr (PullRequest): The created pull request object.
+    """
+    try:
+        # Include the 'resources' directory in the file path
+        file_path = f"resources/{yaml_file}"
+        file_contents = repo.get_contents(file_path)
+        yaml_content = file_contents.decoded_content.decode('utf-8')
+
+        yaml_data = yaml.safe_load(yaml_content)
+
+        # Check the structure of the YAML data
+        if 'resources' in yaml_data and isinstance(yaml_data['resources'], list):
+            resources_list = yaml_data['resources']
+        else:
+            st.error("Unexpected YAML structure: Expected a dictionary with a 'resources' key containing a list.")
+            st.stop()
+
+        new_entry = {
+            'authors': authors,
+            'license': license,
+            'name': name,
+            'description': description,
+            'tags': tags,
+            'type': type_,
+            'url': url
+        }
+
+        resources_list.append(new_entry)
+        yaml_data['resources'] = resources_list
+        new_yaml_content = yaml.safe_dump(yaml_data, allow_unicode=True, sort_keys=False)
+
+        # Create a new branch with a unique name based on the YAML file name and a timestamp
+        base_branch = repo.get_branch("main")
+        timestamp = int(time.time())
+        branch_suffix = f"{yaml_file.split('.')[0]}-{timestamp}"
+        new_branch_name = f"update-{branch_suffix}".replace(' ', '-')
+        repo.create_git_ref(ref=f"refs/heads/{new_branch_name}", sha=base_branch.commit.sha)
+
+        # Commit changes to the new branch
+        repo.update_file(file_path, f"Add new entry", new_yaml_content, file_contents.sha, branch=new_branch_name)
+
+        # Create pull request with custom title and description
+        pr_title = f"Add new training materials request to {yaml_file}"
+        pr_body = f"Added new training materials to {yaml_file}."
+        pr = repo.create_pull(title=pr_title, body=pr_body, head=new_branch_name, base='main')
+        st.success(f"Pull request created: {pr.html_url}")
+    except Exception as e:
+        st.error(f"Failed to update YAML file and create pull request: {e}")
+
+# GitHub authentication
+g, repo = authenticate_with_github()
+
+st.title("GitHub Training Materials Submission")
+
+st.markdown("""
+Welcome to the GitHub Training Materials Submission app!
+
+Use this form to add new training materials to our GitHub repository. 
+Please fill in the details below and click "Submit".
+
+Steps:
+1. Select the YAML file for the type of training material.
+2. Provide the necessary information.
+3. Submit the form to create a pull request.
```

Comment:
```suggestion
Steps:
1. Select the YAML file for the type of training material.
2. Provide the necessary information.
3. Submit the form to create a pull request.
```
I suggest to remove this. It feels like a duplication of the sentence above and form entries below.